[quant][graphmode] Add quantization pattern for quantized::add_relu

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33650 [not4land] debugging
* #33533 [not4land] Temporary workaround to get quantized ops
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34805 [quant][graphmode] add constant prop in finalize to simplify graph
* #34804 [quant][graphmode] Replicate quantize node for prim::If
* #34572 [quant][graphmode] quantization support for aten::add
* #34518 [quant][graphmode] quantization work for prim::If
* #34571 [quant][graphmode][fix] use observed_values_ to check values are observed
* #34414 [quant][graphmode][refactor] insertObservers for Block
* #34411 [quant][graphmode][fix] Insert dequantize before use node
* #34349 [quant][graphmode][fix] preserve the type of original value when inserting dequant node
* #34348 [quant] Add dequantize.tensors
* #34803 [quant][graphmode] quantization support for aten::rehshape
* #34347 [quant][graphmode] Add quantization support for aten::dropout
* #34346 [quant][graphmode] Add quantization support for aten::cat
* #34345 [quant][graphmode] Add prim::ListConstruct to general op handling
* #34315 [quant][graphmode] insert quant/dequant work for duplicated debugName
* #34314 [quant][graphmode][refactor] Change QParamMap to QParamVector
* #33854 [quant][graphmode] Quantization pattern for aten::linear
* **#33532 [quant][graphmode] Add quantization pattern for quantized::add_relu**

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20354880](https://our.internmc.facebook.com/intern/diff/D20354880)

raghuramank100 commented on torch/csrc/jit/passes/quantization_patterns.h:
```diff
@@ -88,13 +88,29 @@ graph(%a_quant, %packed_params, %r_scale, %r_zero_point, %r_dtype, %stride, %pad
         %r_quant = quantized::conv2d_relu(%a_quant, %packed_params, %stride, %padding, %dilation, %groups, %r_scale, %r_zero_point)
         return (%r_quant) )";
 
+  std::string inplace_add_relu = R"(
```
What about non-inplace add relu? Do you need a new pattern for that? Also, can you add a test for this pattern?

jerryzh168 replied:
yes, we need another pattern for non inplace add relu

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20354880](https://our.internmc.facebook.com/intern/diff/D20354880)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20354880](https://our.internmc.facebook.com/intern/diff/D20354880)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20354880](https://our.internmc.facebook.com/intern/diff/D20354880)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20354880](https://our.internmc.facebook.com/intern/diff/D20354880)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add quantization pattern for quantized::add_relu"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20354880](https://our.internmc.facebook.com/intern/diff/D20354880)

[ghstack-poisoned]
